Windows/TestHarnesses/T1543.003_WindowsService/ServiceInstaller.ps1

```diff
@@ -1,740 +1,912 @@
+
+            # Only supply registry key context in the case that the service was successfully created.
+            if (Test-Path -Path $NewServiceRegKey -PathType Container) {
+                $ServiceRegistryKey = $NewServiceRegKey
+            }
+
+            # Calling CreateService will create an empty "ObjectName" value which will cause StartService to fail with an error code of 0x4db (WIN32: 1243 ERROR_SERVICE_NOT_FOUND) - "The specified service does not exist."
+            # Explicitly deleting the "ObjectName" value will resolve this issue.
+            if($null -eq $lpServiceStartName){
+                Remove-ItemProperty -Path $ServiceRegistryKey -Name ObjectName -ErrorAction Ignore 
+            }
+
+            if ($StartService) {
+                Write-Verbose 'Attempting to start the service'
+
+                $StartServiceResult = [AtomicTestHarnesses_T1543_003.ProcessNativeMethods]::StartService(
+                    $ServiceHandle, # hService
+                    0,              # dwNumServiceArgs
+                    [IntPtr]::Zero  # lpServiceArgVectors
+                );$LastError = [ComponentModel.Win32Exception][Runtime.InteropServices.Marshal]::GetLastWin32Error()
+
+                if ($StartServiceResult -eq $False) {
+                    # Close the service handle
+                    $null = [AtomicTestHarnesses_T1543_003.ProcessNativeMethods]::CloseServiceHandle($ServiceHandle)
+
+                    # Close the service control manager handle
+                    $null = [AtomicTestHarnesses_T1543_003.ProcessNativeMethods]::CloseServiceHandle($SCHandle)
+
+                    Write-Error "Failed to start service. Reason: $($LastError.Message) (ErrorCode: 0x$($LastError.NativeErrorCode.ToString('X8')))"
+
+                    return
+                }
+            }
+             # Close the service handle
+            $null = [AtomicTestHarnesses_T1543_003.ProcessNativeMethods]::CloseServiceHandle($ServiceHandle)
+
+            # Close the service control manager handle
+            $null = [AtomicTestHarnesses_T1543_003.ProcessNativeMethods]::CloseServiceHandle($SCHandle)
+           }
+    }
+
+    if (($ServiceType -eq  'KernelDriver') -or ($ServiceType -eq 'FileSystemDriver')){
+        Get-ATHDriverService -ServiceName $ServiceName
+    }
+    else{
+        $ServiceInfo = Get-ItemProperty -Path HKLM:\SYSTEM\CurrentControlSet\Services\$ServiceName\ -ErrorAction Ignore
+        if ($null -ne $ServiceInfo){
+            $TestSuccess = $true
+            $ServiceWMIInstance = Get-CimInstance Win32_Service -Filter "Name = '$ServiceName'" | Select-Object -First 1
+            if($null -eq $ServiceWMIInstance){
+                $ServiceState = 'Stopped'
+            }
+            else{
+                $ServiceState = $ServiceWMIInstance.State
+            }
+        }
+        else{
+            $TestSuccess = $false
+        }
+    
+        [PSCustomObject] @{
```

Comment:
Would it make sense to consolidate the output of user mode and driver services so that the object output isn't different? Unless you think it makes more sense to keep the output distinct, that difference would need to be documented accordingly.